### PR TITLE
Allow to inline internal wrappers in compiler

### LINKED ISSUE #1691: Represent native contracts as interop packages
Anything can be invoked with `contract.Call` interop, but it's not very convenient for frequently used native contracts. Especially given the fact that we don't provide their hashes at the `interop` level (take a look at NeoFS `alphabet` contract for example). As more and more things move from syscalls to native contracts we should make it easier to use them. Most of other toolchains provide wrapper classes and we should probably make additional packages (one package --- one contract) with functions.

It's not really hard, but I'm not sure we can correctly make it without #941 (essentially some interop functions change to regular functions that use `contract.Call` inside (though it'd be nice to inline these calls)). It'd also be nice to autogenerate contract hashes for `interop` use and then check them in tests.

### PR DESCRIPTION
**Pull request by fyrchik**
Depends on #1702 .
Close #1691 .
Base for #941.

Haven't implemented general case, but enough to cover all our needs.
Current restrictions (none of them is a problem for wrappers):
1. Parameter in inline function can't be assigned to.
2. There should be exactly 1 exit point (`return`).
3. No `defer`s in inlined functions.
4. No recursion.

None of them is difficult to fix, but I don't see much value in doing this now without some real use-case.

### CODE REVIEW

File: pkg/compiler/inline_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	t.Run("inline function in inline function parameter", func(t *testing.T) {
+		src := fmt.Sprintf(srcTmpl, `return inline.Sum(inline.SumSquared(1, 2), inline.Sum(3, 4))`)
+		checkCallCount(t, src, 0, 1)
+		eval(t, src, big.NewInt(9+3+4))
+	})
+	t.Run("global name clash", func(t *testing.T) {
+		src := fmt.Sprintf(srcTmpl, `return inline.GetSumSameName()`)
+		checkCallCount(t, src, 0, 1)
+		eval(t, src, big.NewInt(42))
+	})
```
Comment (roman-khimov) [inline pkg/compiler/inline_test.go:89]: How about `return inline.Sum(inline.SumSquared(1, 2), sum(3, 4))`

File: pkg/compiler/codegen.go [hunk 2]
```diff
@@ -194,20 +194,21 @@ func (c *codegen) emitStoreStructField(i int) {
 
 // getVarIndex returns variable type and position in corresponding slot,
 // according to current scope.
-func (c *codegen) getVarIndex(pkg string, name string) (varType, int) {
+func (c *codegen) getVarIndex(pkg string, name string) *varInfo {
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:197]: If it never returns `nil`, why does it return a pointer?

File: pkg/compiler/codegen.go [hunk 3]
```diff
@@ -225,8 +226,12 @@ func getBaseOpcode(t varType) (opcode.Opcode, opcode.Opcode) {
 
 // emitLoadVar loads specified variable to the evaluation stack.
 func (c *codegen) emitLoadVar(pkg string, name string) {
-	t, i := c.getVarIndex(pkg, name)
-	c.emitLoadByIndex(t, i)
+	vi := c.getVarIndex(pkg, name)
+	if vi.tv.Value != nil {
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:230]: And `c.getVarIndex()` can return a `nil`, why don't we check for it here?

File: pkg/compiler/codegen.go [hunk 4]
```diff
@@ -230,6 +230,9 @@ func (c *codegen) emitLoadVar(pkg string, name string) {
 	if vi.tv.Value != nil {
 		c.emitLoadConst(vi.tv)
 		return
+	} else if vi.index == -1 {
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:233]: That's like a magic index, it deserves a constant probably.

File: pkg/compiler/inline_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/compiler/inline_test.go:89]: Added this test, works as expected.

File: pkg/compiler/codegen.go (hunk 2 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:197]: Mostly to keep the interface same as with `varScope.getVarInfo`.

File: pkg/compiler/codegen.go (hunk 3 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:230]: `codegen.getVarIndex` never returns nil, it creates a new local if nothing was found.

File: pkg/compiler/codegen.go (hunk 4 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:233]: Fixed.

File: pkg/compiler/codegen.go (hunk 3 above)
Comment (roman-khimov) [reply pkg/compiler/codegen.go:230]: Well it does invoke `c.scope.vars.getVarInfo(name)` and that can return a `nil`, isn't it?

Comment (fyrchik) [reply pkg/compiler/codegen.go:230]: For the first call there is a `nil` check.
For the second call, the variable for `name` is created on a previous line.

May be we should restructure this to make it more obvious.

**Comment by roman-khimov** [comment]
I fear it might destabilize compiler, let's merge it post-0.93.0.

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/compiler/codegen.go (hunk 4 above)
Comment (roman-khimov) [reply pkg/compiler/codegen.go:233]: Was it updated after the fix? I don't see anything changed here.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**